src/controllers/App.ts

```diff
@@ -37,5 +45,12 @@ export default class App {
 
     window.addEventListener('load', () => this.state.router?.resolve());
     window.addEventListener('hashchange', () => this.state.router?.resolve());
+    window.addEventListener('beforeunload', () => {
+      if (this.state.user && this.state.user.userId && this.state.user.token && this.state.user.refreshToken) {
```

Comment:
if (this.state.user?.userId && this.state.user.token && this.state.user.refreshToken) {

Reply:
fix